src/Common/src/Interop/Unix/System.Net.Security.Native/Interop.Gss.cs

```diff
@@ -0,0 +1,336 @@
+            SafeGssNameHandle desiredName,
+            string password,
+            bool isInitiate,
+            out SafeGssCredHandle outputCredHandle);
+
+        [DllImport(Interop.Libraries.SecurityNative)]
+        internal static extern Status GssReleaseCred(
+            out Status minorStatus,
+            ref IntPtr credHandle);
+
+        [DllImport(Interop.Libraries.SecurityNative)]
+        internal static extern Status GssInitSecContext(
+            out Status minorStatus,
+            SafeGssCredHandle initiatorCredHandle,
+            ref SafeGssContextHandle contextHandle,
+            bool isNtlm,
+            SafeGssNameHandle targetName,
+            uint reqFlags,
+            byte[] inputBytes,
+            int inputLength,
+            SafeGssBufferHandle outputToken,
+            out uint retFlags);
+
+        [DllImport(Interop.Libraries.SecurityNative)]
+        internal static extern Status GssAcceptSecContext(
+            out Status minorStatus,
+            ref SafeGssContextHandle contextHandle,
+            SafeGssCredHandle initiatorCredHandle,
+            SafeGssBufferHandle inputToken,
+            SafeGssBufferHandle outputToken,
+            out uint retFlags);
+
+        [DllImport(Interop.Libraries.SecurityNative)]
+        internal static extern Status GssDeleteSecContext(
+            out Status minorStatus,
+            ref IntPtr contextHandle);
+
+        [DllImport(Interop.Libraries.SecurityNative)]
+        internal static extern Status GssWrap(
+            out Status minorStatus,
+            SafeGssContextHandle contextHandle,
+            bool isEncrypt,
+            byte[] inputBytes,
+            int offset,
+            int count,
+            SafeGssBufferHandle outputMessageBuffer);
+
+        [DllImport(Interop.Libraries.SecurityNative)]
+        internal static extern Status GssUnwrap(
+            out Status minorStatus,
+            SafeGssContextHandle contextHandle,
+            byte[] inputBytes,
+            int offset,
+            int count,
+            SafeGssBufferHandle outputMessageBuffer);
+
+        [DllImport(Interop.Libraries.SecurityNative)]
+        internal static extern Status GssInquireSourceName(
+            out Status minorStatus,
+            SafeGssContextHandle contextHandle,
+            out SafeGssNameHandle srcName);
+
+        [DllImport(Interop.Libraries.SecurityNative)]
+        internal static extern Status GssDisplayName(
+            out Status minorStatus,
+            SafeGssNameHandle inputName,
+            SafeGssBufferHandle outputNameBuffer);
+
+        [StructLayout(LayoutKind.Sequential)]
+        internal struct gss_buffer_desc
+        {
+            internal size_t length;
+            internal IntPtr value;
+        }
+
+        internal enum Status : uint
+        {
+            GSS_S_COMPLETE = 0,
+            GSS_S_CONTINUE_NEEDED = 1
+        }
+
+        [FlagsAttribute]
+        internal enum GssFlags : uint
```

Comment:
This enum should be implemented using the same pattern as the other "PAL" enums.
1. It should be defined in the shim's .h file.
2. There should be "static_assert" lines for each value in the shim's .cpp file to ensure our value here always matches the value used in the library we are compiling against.

See 
* https://github.com/dotnet/corefx/blob/master/src/Native/System.Net.Http.Native/pal_easy.h#L70-L76
* https://github.com/dotnet/corefx/blob/master/src/Native/System.Net.Http.Native/pal_easy.cpp#L44-L46
* https://github.com/dotnet/corefx/blob/master/src/Common/src/Interop/Unix/System.Net.Http.Native/Interop.Easy.cs#L123-L129